### PR TITLE
Adds slash command for yogalog

### LINKED ISSUE #6: Create Slash Command to list log
https://discordjs.guide/slash-commands/response-methods.html

`/yoga` should give the number of yoga days, and maybe days per month?

Should is also be possible to get all sessions of a week, with the notes?

What do you think @davidwessman?

### PR DESCRIPTION
**Pull request by davidwessman**
- Lists all recorded sessions for the user
- Needs to deploy commands to servers
- Followed this guide: https://discordjs.guide/creating-your-bot/slash-commands.html
- Fixes #6 

Screenshot 2023-02-04 at 17 26 06
Screenshot 2023-02-04 at 17 26 16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,10 @@ Write a log message in the #yoga channel.
 - Install dependencies `yarn`
 - Start the bot `yarn dev`
 
+## 🤖 How to deploy
+
+- Need to run `node ./src/deploy-commands.js` to deploy the slash commands to all connected Discord servers.
```
Comment (davidwessman) [inline README.md]: Github Copilot suggested the title, and even though there are no other deploy-instructions I thought it looked nice 🙂

File: src/config.js [hunk 2]
```diff
@@ -2,3 +2,4 @@ import * as dotenv from "dotenv";
 dotenv.config();
 
 export const DISCORD_CLIENT_TOKEN = process.env.DISCORD_CLIENT_TOKEN;
+export const DISCORD_CLIENT_ID = process.env.DISCORD_CLIENT_ID;
```
Comment (davidwessman) [inline src/config.js:5]: Needed when deploying the commands

File: src/commands/yogalog.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { SlashCommandBuilder } from "discord.js";
+import { getXataClient } from "./../xata.js";
+
+export default {
+  data: new SlashCommandBuilder()
+    .setName('yogalog')
+    .setDescription('Lists all your yoga sessions'),
+  async execute(interaction, xata = getXataClient()) {
```
Comment (raae) [inline src/commands/yogalog.js:28]: **question(non-blocking):** is this the same as `execute: (interaction, xata = getXataClient()) => {}`?

File: src/commands/yogalog.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    const discordUserId = interaction.user.id;
+
+    const results = await xata.db.session
+      .filter({discordUserId: discordUserId})
+      .getMany();
```
Comment (raae) [inline src/commands/yogalog.js]: **suggestion(non-blocking):** use getAll()?

**Review by raae — APPROVED** [review]
**question(non-blocking):** Could we do deploy the commands as a part of the start script, then the railway service would do it and we'd only need the prod keys in one place 🤔

File: src/commands/yogalog.js (hunk 3 above)
Comment (raae) [reply src/commands/yogalog.js:28]: I saw they do this in the docs, just very unfamiliar to with a function declaration with await in the middle of a object declaration.

**Comment by davidwessman** [comment]
I will merge like this and create a new PR with improvements 🙂

File: src/commands/yogalog.js (hunk 3 above)
Comment (davidwessman) [reply src/commands/yogalog.js:28]: Hm, I am not sure which syntax is more expressive, but I had to do this to use `await` inside it.

**Comment by davidwessman** [comment]
@raae Seems I cannot merge myself, but I did some changes based on your feedback 🙂

**Comment by davidwessman** [comment]
Managed to make some filtering work 🙂 #
https://user-images.githubusercontent.com/6763624/216833817-e75164c0-4d06-4c99-9e6b-1bc364f2b2d3.mov

**Comment by raae** [comment]
**praise:** I like the filtering!

File: src/commands/yogalog.js (hunk 3 above)
Comment (raae) [reply src/commands/yogalog.js:28]: I forgot the async in my question: `execute: async (interaction, xata = getXataClient()) => {}`.

**Comment by davidwessman** [comment]
@raae Will you merge it? 😊

**Comment by raae** [comment]
Yes!

File: src/commands/yogalog.js (hunk 3 above)
Comment (davidwessman) [reply src/commands/yogalog.js:28]: Hm, other default keys work that way - but functions? Not sure.

**Pull request merged**